tests/unit/synapseclient/unit_test_client.py

```diff
@@ -767,6 +767,46 @@ def test_move():
         syn_store_patch.assert_called_once_with(moved_entity, forceVersion=False)
 
 
+def test_delete__bad_attribute():
+    assert_raises(SynapseError, syn.delete, ["foo"])
+
+
+def test_delete__string():
+    with patch.object(syn, "restDELETE") as patch_rest_delete:
+        syn.delete("syn1235")
+        patch_rest_delete.assert_called_once_with(uri="/entity/syn1235")
+
+
+def test_delete__string_version():
+    with patch.object(syn, "restDELETE") as patch_rest_delete:
+        syn.delete("syn1235", version=1)
+        patch_rest_delete.assert_called_once_with(uri="/entity/syn1235/version/1")
+
+
+class mock_synapse_delete:
```

Comment:
Remove this class and just pass in a new `mock_obj=MagicMock()` object in the test.

has_attr() will always return true for MagicMock

you can then assert that was called `mock_obj._synapse_delete.assert_called_once()`